app/src/main/java/com/tiptopgoodstudio/androidresources/ui/adapters/ResourceListAdapter.java

```diff
@@ -51,33 +41,34 @@ public ResourceListAdapterViewHolder onCreateViewHolder(ViewGroup parent, int vi
         Context context = parent.getContext();
         LayoutInflater inflater = LayoutInflater.from(context);
         View itemView = inflater.inflate(R.layout.resource_item, parent, false);
-        ResourceListAdapterViewHolder viewHolder = new ResourceListAdapterViewHolder(itemView);
-        return viewHolder;
+        return new ResourceListAdapterViewHolder(itemView);
     }
 
     @Override
     public void onBindViewHolder(ResourceListAdapterViewHolder holder, int position) {
         Resource currentResource = mResourceData.get(position);
 
-        holder.itemView.setTag(currentResource.getResourceId());
-
         holder.mResourceUrl.setText(currentResource.getResourceUrl());
-
         holder.mResourceDescription.setText(currentResource.getResourceDescription());
 
         String resourceFormat = currentResource.getResourceFormat();
         int resourceId = R.drawable.resource_url_icon;
-        if(resourceFormat == null) {
-            //Default it to url
-            resourceFormat = "url";
-        }
-        else {
-            if(resourceFormat.equalsIgnoreCase("pdf")) {
-                resourceId = R.drawable.resource_pdf_icon;
-            } else if(resourceFormat.equalsIgnoreCase("video")) {
-                resourceId = R.drawable.resource_video_icon;
-            } else if(resourceFormat.equalsIgnoreCase("image")) {
-                resourceId = R.drawable.resource_image_icon;
+        if (resourceFormat == null) {
+            resourceFormat = RESOURCE_TYPE_PDF;
+        } else {
+            switch (resourceFormat) {
+                case RESOURCE_TYPE_PDF:
+                    resourceId = R.drawable.resource_pdf_icon;
+                    break;
+                case RESOURCE_TYPE_VIDEO:
+                    resourceId = R.drawable.resource_video_icon;
+                    break;
+                case RESOURCE_TYPE_IMAGE:
+                    resourceId = R.drawable.resource_image_icon;
+                    break;
+                default:
+                    resourceId = R.drawable.resource_pdf_icon;
```

Comment:
Even in the switch statement, the default needs to be R.drawable.resource_url_icon